c++/src/Compression.cc

```diff
@@ -232,6 +244,22 @@ namespace orc {
       }
       rawInputBuffer.resize(0);
     }
+  }
+
+  bool CompressionStream::Next(void** data, int* size) {
+    if (rawInputBuffer.size() > compressionBlockSize) {
+      std::stringstream ss;
+      ss << "uncompressed data size " << rawInputBuffer.size() << " is larger than block size "
+         << compressionBlockSize
+         << ". compressionBlockSize should be set equal to multiply of "
+            "memoryBlockSize";
+      throw std::logic_error(ss.str());
```

Comment:
Should we add a new Exception type?

Reply:
Maybe we can replace all std::logic_error in CompressionStream with new exception types.  I'll address this issue in my next work item.